### DISCUSSION TITLE
Load gopher-lua libraries at runtime

### DISCUSSION THREAD

**Discussion by asumbek** [yuin/gopher-lua#436] [Q&A] [answered]
Is it possible to load gopher-lua libraries at runtime? If not is there a workaround for it (maybe compiling it as shared libraries and load it via `dlopen` or other alternatives at runtime)?

**Comment by cloudwindy** [comment, marked as answer]
You can achieve that by go [plugin](https://pkg.go.dev/plugin).

**Comment by cloudwindy** [comment]
But go plugins are not suitable for most of the times because of many problems in its design. You can look at [Things to avoid while using Golang plugins](https://alperkose.medium.com/things-to-avoid-while-using-golang-plugins-f34c0a636e8). It basically only "allows" you to load your plugins but they are not very flexible.
